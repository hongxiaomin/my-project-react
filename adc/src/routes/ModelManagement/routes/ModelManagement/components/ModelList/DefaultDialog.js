import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { defaultModel } from '../../modules'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  isDefaultOpen: Boolean,
  toggleDefault: Function,
  dispatch: Function,
};

export class DefaultDialog extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    const tmp = {
      modelId: this.props.dataDetail.original.id,
      asDefault: true,
    }
    console.log('default', tmp)
    this.props.dispatch(defaultModel(tmp))
    this.props.toggleDefault()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.isDefaultOpen}
            onClose={this.props.toggleDefault}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}><WrapperFormattedMessage id="confirmation" /></h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>
                <WrapperFormattedMessage id="modelManagement.defaultDialog.note" />
              </p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleDefault}>
                  <WrapperFormattedMessage id="cancel" />
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  <WrapperFormattedMessage id="Save" />
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(DefaultDialog)
