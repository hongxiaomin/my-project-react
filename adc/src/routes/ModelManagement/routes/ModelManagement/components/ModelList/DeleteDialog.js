import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { deleteModel } from '../../modules'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  delOpen: Boolean,
  toggleDel: Function,
  dispatch: Function
};

export class DeleteDialog extends React.Component {


  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    this.props.dispatch(deleteModel(this.props.dataDetail.original.id))
    this.props.toggleDel()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.delOpen}
            onClose={this.props.toggleDel}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}><WrapperFormattedMessage id="confirmation" /></h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>
                <WrapperFormattedMessage id="modelManagement.deleteDialog.note" />
              </p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleDel}>
                  <WrapperFormattedMessage id="cancel" />
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  <WrapperFormattedMessage id="Save" />
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(DeleteDialog)
