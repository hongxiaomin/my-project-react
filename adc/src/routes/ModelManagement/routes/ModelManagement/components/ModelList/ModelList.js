import React from 'react'
import { isEmpty } from 'lodash'
import { connect } from 'react-redux'
import ReactTable from 'components/ADCTable'
import { Icon } from 'antd'
import { injectIntl } from 'react-intl'
import { modelListStatus } from 'Utils'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import Delete from './DeleteDialog'
import Publish from './PublishDialog'
import UnPublish from './UnPublishDialog'
import DefaultD from './DefaultDialog'
import UnDefault from './UnDefaultDialog'
import ImportModel from '../AddImportModel'
// import EditModel from '../EditModel'
import EditModelName from '../EditModelName'
import ActionModel from '../ActionModel'
import classes from './ModelList.scss'
import { getModelList, PAGE_STATUS, getModelDetail, COPY_MODEL_DETAIL } from '../../modules'

type Props = {
  dispatch: Function,
  modelList: Array,
  isDetail: Function,
  intl: Object,
}

const activeRowStyle = {
  background: 'skyblue',
  color: '#fff',
  height: '35px',
  lineHeight: '20px',
}

export class ModelList extends React.Component {
  constructor() {
    super()
    this.state = {
      highlightedIndex: -1,
      isOpen: false,
      editOpen: false,
      editNameOpen: false,
      copyOpen: false,
      delOpen: false,
      publishOpen: false,
      unPublishOpen: false,
      importOpen: false,
      isDefaultOpen: false,
      unDefaultOpen: false,
      dataDetail: {},
    }
    this.handleClick = this.handleClick.bind(this)
    this.del = this.del.bind(this)
    this.toggleEdit = this.toggleEdit.bind(this)
    this.toggleCopy = this.toggleCopy.bind(this)
    this.toggleEditName = this.toggleEditName.bind(this)
    this.togglePublish = this.togglePublish.bind(this)
    this.toggleImport = this.toggleImport.bind(this)
    this.toggleUnPublish = this.toggleUnPublish.bind(this)
    this.toggleunDefault = this.toggleunDefault.bind(this)
    this.toggleDefault = this.toggleDefault.bind(this)
  }

  componentDidMount() {
    this.props.dispatch(getModelList())
  }

  handleClick(data) {
    this.setState({
      isOpen: true,
    })
    this.props.isDetail(true, data)
    this.props.dispatch(getModelDetail(data._original.id)) // 传入modelid
    this.props.dispatch({
      type: PAGE_STATUS,
      payload: 'detailPage',
    })
  }
  toggleImport() {
    this.setState({
      importOpen: !this.state.importOpen,
    })
  }
  toggleDefault(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        isDefaultOpen: !this.state.isDefaultOpen,
      })
    }
  }
  toggleunDefault(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        unDefaultOpen: !this.state.unDefaultOpen,
      })
    }
  }
  toggleEdit(data) {
    return () => {
      let tmp
      if (data && data.original) {
        tmp = {
          ...data.original,
        }
      }
      this.setState({
        ...this.state,
        dataDetail: tmp,
        editOpen: !this.state.editOpen,
      }, () => {
        this.props.dispatch({
          type: COPY_MODEL_DETAIL,
          data: this.state.dataDetail,
        })
      })
    }
  }
  toggleCopy(data) {
    return () => {
      let tmp
      if (data && data.original) {
        tmp = {
          ...data.original,
        }
      }
      this.setState({
        ...this.state,
        dataDetail: tmp,
        copyOpen: !this.state.copyOpen,
      }, () => {
        this.props.dispatch({
          type: COPY_MODEL_DETAIL,
          data: this.state.dataDetail,
        })
      })
    }
  }
  toggleEditName(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        editNameOpen: !this.state.editNameOpen,
      })
    }
  }
  togglePublish(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        publishOpen: !this.state.publishOpen,
      })
    }
  }
  toggleUnPublish(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        unPublishOpen: !this.state.unPublishOpen,
      })
    }
  }
  del(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        delOpen: !this.state.delOpen,
      })
    }
  }

  actionCell = intl => data => (
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
      {
        data.original.status !== 'Idle' ?
          <button
            type='button'
            onClick={this.toggleEditName(data)}
            className={'pt-button pt-minimal pt-icon-edit'}
            title={intl.formatMessage({ id: 'edit' })}
          /> :
          <button
            type='button'
            onClick={this.toggleEdit(data)}
            className={'pt-button pt-minimal pt-icon-edit'}
            title={intl.formatMessage({ id: 'edit' })}
          />
      }
      <button
        type='button'
        onClick={this.toggleCopy(data)}
        className={'pt-button pt-minimal pt-icon-duplicate'}
        title={intl.formatMessage({ id: 'copy' })}
      />
      <button
        type='button'
        onClick={this.del(data)}
        disabled={data.original.status === 'activated'}
        className={'pt-button pt-minimal pt-icon-delete'}
        title={intl.formatMessage({ id: 'delete' })}
      />
      {
        data.original.isPublished ?
          <button
            type='button'
            onClick={this.toggleUnPublish(data)}
            disabled={data.original.status !== 'Tested'}
            className={'pt-button pt-minimal pt-icon-cloud-download'}
            title={intl.formatMessage({ id: 'inactivate' })}
          /> :
          <button
            type='button'
            onClick={this.togglePublish(data)}
            disabled={data.original.status !== 'Tested'}
            className={'pt-button pt-minimal pt-icon-cloud-upload'}
            title={intl.formatMessage({ id: 'activate' })}
          />
      }
      {
        data.original.isDefault ?
          <button
            type='button'
            onClick={this.toggleunDefault(data)}
            disabled={data.original.isPublished !== true}
            className={'pt-button pt-minimal'}
            title={intl.formatMessage({ id: 'modelManagement.modelList.cancelDefaultModel' })}
          >
            <Icon type="star" style={{ color: 'orange' }} />
          </button> :
          <button
            type='button'
            onClick={this.toggleDefault(data)}
            disabled={data.original.isPublished !== true}
            className={'pt-button pt-minimal'}
            title={intl.formatMessage({ id: 'modelManagement.modelList.setDefaultModel' })}
          >
            <Icon type="star" />
          </button>
      }
    </div>
  )

  props: Props

  render() {
    const { intl } = this.props
    const style = { textAlign: 'center' }
    const columns = [
      {
        Header: `${intl.formatMessage({ id: 'modelId' })}`,
        accessor: 'id',
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'modelName' })}`,
        accessor: 'name',
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'status' })}`,
        accessor: 'status',
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'service' })}`,
        id: 'currentState',
        accessor: d => ((d.currentState === 'none' && d.isTrained === true && d.isVerified === false) ||
        (d.currentState === 'none' && d.isTrained === false && d.isVerified === false) ||
        (d.currentState === 'none' && d.isTrained === true && d.isVerified === true) ? 'Not in use' : 'In use'),
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'modelManagement.modelList.numberOfIteration' })}`,
        accessor: 'iteration',
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'lossRate' })}`,
        accessor: 'lossRate',
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'modelVersion' })}`,
        accessor: 'version',
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'modelManagement.ModelDetail.createTime' })}`,
        accessor: 'createTime',
        style,
      },
      {
        Header: `${intl.formatMessage({ id: 'action' })}`,
        accessor: '',
        style,
        Cell: this.actionCell(intl),
      },
    ]

    return (
      <div>
        <div className={classes['top-left']}>
          <span className={classes.productGlass}>
            <WrapperFormattedMessage id="modelManagement.modelList" />
          </span>
          <ImportModel />
        </div>
        <ReactTable
          data={modelListStatus(this.props.modelList)}
          defaultPageSize={5}
          className="-striped -highlight"
          style={{ height: '255px' }}
          columns={columns}
          getTrProps={(...params) => {
            const rowInfo = params[1]
            if (rowInfo) {
              return {
                style: (this.state.selectedIndex === rowInfo.index) ? activeRowStyle : { height: '35px', lineHeight: '20px' },
                onClick: () => {
                  this.setState({
                    selectedIndex: rowInfo.index,
                  })
                  this.handleClick(rowInfo.row)
                },
              }
            }
            return {}
          }}
        />
        <Delete
          toggleDel={this.del()}
          dataDetail={this.state.dataDetail}
          delOpen={this.state.delOpen}
        />
        <DefaultD
          toggleDefault={this.toggleDefault()}
          dataDetail={this.state.dataDetail}
          isDefaultOpen={this.state.isDefaultOpen}
        />
        <UnDefault
          toggleunDefault={this.toggleunDefault()}
          dataDetail={this.state.dataDetail}
          unDefaultOpen={this.state.unDefaultOpen}
        />
        <Publish
          togglePublish={this.togglePublish()}
          dataDetail={this.state.dataDetail}
          publishOpen={this.state.publishOpen}
        />
        <UnPublish
          toggleUnPublish={this.toggleUnPublish()}
          dataDetail={this.state.dataDetail}
          unPublishOpen={this.state.unPublishOpen}
        />
        {
          !isEmpty(this.state.dataDetail) &&
          <ActionModel
            title={intl.formatMessage({ id: 'modelManagement.actionModel.editModel' })}
            toggle={this.toggleEdit()}
            visible={this.state.editOpen}
          />
        }
        {
          !isEmpty(this.state.dataDetail) &&
          <ActionModel
            title={intl.formatMessage({ id: 'modelManagement.actionModel.copyModel' })}
            toggle={this.toggleCopy()}
            visible={this.state.copyOpen}
          />
        }
        {
          !isEmpty(this.state.dataDetail) &&
          this.state.editNameOpen &&
          <EditModelName
            toggleEdit={this.toggleEditName()}
            dataDetail={this.state.dataDetail}
            editOpen={this.state.editNameOpen}
          />
        }
      </div>
    )
  }
}

export default injectIntl(connect(
  state => ({
    token: state.auth.token,
    modelList: state.modelManagement.modelList,
  }),
)(ModelList))
