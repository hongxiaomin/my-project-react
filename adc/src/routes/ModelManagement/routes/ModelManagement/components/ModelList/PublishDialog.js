import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { actionModel } from '../../modules'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  publishOpen: Boolean,
  togglePublish: Function,
  dispatch: Function,
};

export class PublishDialog extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    const tmp = {
      modelId: this.props.dataDetail.original.id,
      actionType: 'publish',
    }
    console.log('publish', tmp)
    this.props.dispatch(actionModel(tmp))
    this.props.togglePublish()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.publishOpen}
            onClose={this.props.togglePublish}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}><WrapperFormattedMessage id="confirmation" /></h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>
                <WrapperFormattedMessage id="modelManagement.publishDialog.not" />
              </p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.togglePublish}>
                  <WrapperFormattedMessage id="cancel" />
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  <WrapperFormattedMessage id="Save" />
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(PublishDialog)
