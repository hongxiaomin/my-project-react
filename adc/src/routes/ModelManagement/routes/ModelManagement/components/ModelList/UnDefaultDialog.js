import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { defaultModel } from '../../modules'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  unDefaultOpen: Boolean,
  toggleunDefault: Function,
  dispatch: Function,
};

export class UnDefaultDialog extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    const tmp = {
      modelId: this.props.dataDetail.original.id,
      asDefault: false,
    }
    console.log('undefault', tmp)
    this.props.dispatch(defaultModel(tmp))
    this.props.toggleunDefault()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.unDefaultOpen}
            onClose={this.props.toggleunDefault}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}><WrapperFormattedMessage id="confirmation" /></h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>
                <WrapperFormattedMessage id="modelManagement.unDefaultDialog.note" />
              </p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleunDefault}>
                  <WrapperFormattedMessage id="cancel" />
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  <WrapperFormattedMessage id="Save" />
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(UnDefaultDialog)
