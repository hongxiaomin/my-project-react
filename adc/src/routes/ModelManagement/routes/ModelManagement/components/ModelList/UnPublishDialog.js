import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { actionModel } from '../../modules'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  unPublishOpen: Boolean,
  toggleUnPublish: Function,
  dispatch: Function,
};

export class UnPublishDialog extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    const tmp = {
      modelId: this.props.dataDetail.original.id,
      actionType: 'deprecate',
    }
    console.log('unPublishOpen', tmp)
    this.props.dispatch(actionModel(tmp))
    this.props.toggleUnPublish()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.unPublishOpen}
            onClose={this.props.toggleUnPublish}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}><WrapperFormattedMessage id="confirmation" /></h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>
                <WrapperFormattedMessage id="modelManagement.unPublishDialog.note" />
              </p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleUnPublish}>
                  <WrapperFormattedMessage id="cancel" />
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  <WrapperFormattedMessage id="Save" />
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(UnPublishDialog)
