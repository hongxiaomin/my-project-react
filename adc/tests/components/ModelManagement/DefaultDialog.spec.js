import React from 'react'
import { shallow } from 'enzyme'
import { Button } from 'antd'
import { DefaultDialog } from 'routes/ModelManagement/routes/ModelManagement/components/ModelList/DefaultDialog'

describe('(Component) DefaultDialog', () => {
  const dispatch = jest.fn()
  const toggleDefault = jest.fn()
  const dataDetail = {
    original: {
      id: 2,
    },
  }
  const wrapper = shallow(
    <DefaultDialog
      dataDetail={dataDetail}
      dispatch={dispatch}
      toggleDefault={toggleDefault}
      isDefaultOpen={false}
    />,
  )
  it('render normally', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('handleSubmit is called', () => {
    wrapper.find('Button').last().simulate('click')
  })
})
