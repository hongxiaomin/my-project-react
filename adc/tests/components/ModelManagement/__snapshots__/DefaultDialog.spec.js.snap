// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DefaultDialog render normally 1`] = `
<div
  className="Dialog-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        Confirmation
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        Do you want to set this model as default?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;
